### DISCUSSION TITLE
Recommended way to handle deployments where your production API doesn't have introspection enabled?

### DISCUSSION THREAD

**Discussion by watzon** [Diizzayy/nuxt-graphql-client#367] [Q&A]
We're using this for an application which connects to a staging application in development, but in production it needs to connect to the production version of the app. The staging version of the app has introspection turned on, as it isn't public facing, but the production app has introspection turned off seeing as it isn't a public API and we'd rather not have people skimming our schema. This means that when doing a build in the production environment codegen doesn't work and the build inevitably fails.

Is there a recommended pipeline for local development to production deployment with this?

**Comment by Diizzayy** [comment]
@watzon The code generation can be disabled in production as [seen here](https://nuxt-graphql-client.web.app/advanced/codegen#disable-codegen). Perhaps you can have it disabled based on the environment variable NODE_ENV being set to `production`

**Reply by watzon** [reply]
I saw that, but without the codegen in production wouldn't I have to do something before staging to make sure that generated code is still available?
